easybuild/framework/easyblock.py

```diff
@@ -2898,7 +2901,21 @@ def build_and_install_one(ecdict, init_env):
                 app.cfg.dump(reprod_spec)
                 _log.info("Dumped fully processed easyconfig to %s", reprod_spec)
             except NotImplementedError as err:
-                _log.warn("Unable to dumped fully processed easyconfig to %s: %s", reprod_spec, err)
+                _log.warn("Unable to dump fully processed easyconfig to %s: %s", reprod_spec, err)
+            # also archive the relevant easyblocks
+            reprod_spec = os.path.join(new_log_dir, 'reprod', 'easyblocks')
+            for easyblock_class in inspect.getmro(app_class):
+                easyblock_path = inspect.getsourcefile(easyblock_class)
+                easyblock_basedir, easyblock_filename = os.path.split(easyblock_path)
+                # if we reach the framework modules we are done
+                if easyblock_filename in FRAMEWORK_EASYBLOCK_MODULES:
```

Comment:
@ocaisa I think it's better to check this via `easyblock_class`, against `EasyBlock` and `ExtensionEasyblock`, rather than relying on filenames of the Python modules.

For `ExtensionEasyblock` that as a bit tricky though, since we can't import that in here (it would lead to a cyclic import). So, how about this:

```python
for easyblock_class in inspect.getmro(app_class):
    # if we reach EasyBlock or ExtensionEasyblock class, we are done
    if easyblock_class.__name__ in [EasyBlock.__name__, 'ExtensionEasyblock']:
        break
    ...
```

And then chuck `FRAMEWORK_EASYBLOCK_MODULES`, and hardcode `easyblock.py` and `extensioneasyblock.py` in the tests (it's OK there imho, and then we're checking indirectly compared to how this is actually implemented).